lib/ex_aws/dynamo/encodable.ex

```diff
@@ -111,3 +111,29 @@ defimpl ExAws.Dynamo.Encodable, for: List do
     %{"L"  => typed_values}
   end
 end
+
+defimpl ExAws.Dynamo.Encodable, for: MapSet do
+  alias ExAws.Dynamo.Encodable
+
+  def encode(mapset, _) do
+    cond do
+      MapSet.size(mapset) == 0 ->
+        raise "Cannot determine a proper data type for an empty MapSet"
+      Enum.all?(mapset, &is_number/1) ->
+        %{"NS" => number_set_to_list(mapset)}
+      Enum.all?(mapset, &is_binary/1) ->
+        %{"SS" => MapSet.to_list(mapset)}
+      true ->
+        raise "All elements in a MapSet must be only numbers or only strings"
```

Comment:
I'm not wildly enthusiastic about raising exceptions here, but i'm not sure that the alternatives are much better.